cli/config.go

```diff
@@ -30,8 +30,14 @@ func (c *App) onRepositoryFatalError(f func(err error)) {
 	c.onFatalErrorCallbacks = append(c.onFatalErrorCallbacks, f)
 }
 
+func (c *App) onDebugDump(f func()) {
+	onSig(c.simulatedSigDump, SignalDump, f)
+}
+
 func (c *App) onTerminate(f func()) {
-	s := make(chan os.Signal, 1)
+	// channel length of 2 to accommodate both signals
```

Comment:
there's a note in `signal.c`

Reply:
As you point out, this is a single-shot handler, so the rest of the comments below are not relevant to the change, mainly for discussion purposes.

This callback (channel) is only installed for `SIGTERM` and `SIGINT`.
`SIGKILL` is not interceptable as you know.

> ... _"has sufficient buffer space to keep up with the expected signal rate"_ ...

They way I understand the documentation you referenced above for `signal.Notify`, is that:
- The channel must be buffered (that is, buffer size > 0). If the channel is not buffered, then the message will not be added to the channel because the signal handler will not block on sending to the channel. That is, if the channel is unbuffered or full, then it will simply not send the message on the channel.
- If the routine reading from the channel cannot read fast enough to keep up with the signal rate, then messages will not be delivered to the channel, so the channel buffer size should be proportional to the maximum signal rate that is to be handled.